models/service_test.go

```diff
@@ -0,0 +1,205 @@
+package models
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/swift-sunshine/swscore/kubernetes"
+	"github.com/swift-sunshine/swscore/models"
+
+	"k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type prometheusClientMock struct {
+	mock.Mock
+}
+
+type IstioClientMock struct {
+	mock.Mock
+}
+
+func (c *IstioClientMock) GetServiceDetails(namespaceName, serviceName string) (*kubernetes.ServiceDetails, error) {
+	args := c.Called(namespaceName, serviceName)
+	return args.Get(0).(*kubernetes.ServiceDetails), args.Error(1)
+}
+
+func (c *IstioClientMock) GetIstioDetails(namespaceName, serviceName string) (*kubernetes.IstioDetails, error) {
+	args := c.Called(namespaceName, serviceName)
+	return args.Get(0).(*kubernetes.IstioDetails), args.Error(1)
+}
+
+func (c *prometheusClientMock) GetSourceServices(namespaceName, serviceName string) (map[string][]string, error) {
+	args := c.Called(namespaceName, serviceName)
+	return args.Get(0).(map[string][]string), args.Error(1)
+}
+
+func mockServiceDetails() *IstioClientMock {
+	mockIstioClient := new(IstioClientMock)
+
+	serviceDetails := fakeServiceDetails()
+	mockIstioClient.On("GetServiceDetails", "namespace", "service").Return(serviceDetails, nil)
+
+	istioDetails := fakeIstioDetails()
+	mockIstioClient.On("GetIstioDetails", "namespace", "service").Return(istioDetails, nil)
+
+	return mockIstioClient
+}
+
+func mockPrometheusDetails() *prometheusClientMock {
+	mockPrometheusClient := new(prometheusClientMock)
+	prometheusDetails := fakePrometheusDetails()
+	mockPrometheusClient.On("GetSourceServices", "namespace", "service").Return(prometheusDetails, nil)
+
+	return mockPrometheusClient
+}
+
+func TestServiceDetailParsing(t *testing.T) {
+	mockIstioClient := mockServiceDetails()
+	mockPrometheusClient := mockPrometheusDetails()
+
+	service := models.Service{}
+	service.Name = "service"
+	service.Namespace = models.Namespace{"namespace"}
+	service.SetIstioClient(mockIstioClient)
+	service.SetPrometheusClient(mockPrometheusClient)
+	err := service.SetServiceDetails()
+	if err != nil {
+		assert.Equal(t, err, nil, "Error found during parsing")
+	}
+
+	mockIstioClient.AssertExpectations(t)
+	mockPrometheusClient.AssertExpectations(t)
+
+	// Kubernetes Details
+	assert.Equal(t, service.Name, "service")
+	assert.Equal(t, service.Namespace.Name, "namespace")
+	assert.Equal(t, service.Type, "ClusterIP")
+	assert.Equal(t, service.Ip, "fromservice")
+	assert.Equal(t, service.Labels, map[string]string{"label1": "labelName1", "label2": "labelName2"})
+	assert.Equal(t, service.Ports, models.Ports{
```

Comment:
there are more assertion related functions to compare complex structures. did you consider using them?